pkg/apis/k0s.k0sproject.io/v1beta1/api.go

```diff
@@ -52,11 +53,12 @@ func DefaultAPISpec() *APISpec {
 	addresses, _ := iface.AllAddresses()
 	publicAddress, _ := iface.FirstPublicAddress()
 	return &APISpec{
-		Port:       6443,
-		K0sAPIPort: 9443,
-		SANs:       addresses,
-		Address:    publicAddress,
-		ExtraArgs:  make(map[string]string),
+		Port:                   6443,
+		K0sAPIPort:             9443,
+		SANs:                   addresses,
+		Address:                publicAddress,
+		ExtraArgs:              make(map[string]string),
+		TunneledNetworkingMode: true,
```

Comment:
Making it default to `true` means that once user upgrades, cluster will switch to this mode. Maybe we should disable it by default?